parser/errors.go

```diff
@@ -359,37 +393,1193 @@ func (e *MissingCommaInParameterListError) StartPosition() ast.Position {
+
+func (e *DuplicateExecuteBlockError) StartPosition() ast.Position {
+	return e.Pos
+}
+
+func (e *DuplicateExecuteBlockError) EndPosition(_ common.MemoryGauge) ast.Position {
+	return e.Pos
+}
+
+func (e *DuplicateExecuteBlockError) Error() string {
+	return fmt.Sprintf(
+		"unexpected second %q block",
+		KeywordExecute,
+	)
+}
+
+func (*DuplicateExecuteBlockError) SecondaryError() string {
+	return "transaction declarations can only have one 'execute' block"
+}
+
+func (*DuplicateExecuteBlockError) DocumentationLink() string {
+	return "https://cadence-lang.org/docs/language/transactions"
+}
+
+// DuplicatePostConditionsError is reported when a transaction declaration has a second post-conditions block.
+type DuplicatePostConditionsError struct {
+	Pos ast.Position
+}
+
+var _ ParseError = &DuplicatePostConditionsError{}
+var _ errors.UserError = &DuplicatePostConditionsError{}
+var _ errors.SecondaryError = &DuplicatePostConditionsError{}
+var _ errors.HasDocumentationLink = &DuplicatePostConditionsError{}
+
+func (*DuplicatePostConditionsError) isParseError() {}
+
+func (*DuplicatePostConditionsError) IsUserError() {}
+
+func (e *DuplicatePostConditionsError) StartPosition() ast.Position {
+	return e.Pos
+}
+
+func (e *DuplicatePostConditionsError) EndPosition(_ common.MemoryGauge) ast.Position {
+	return e.Pos
+}
+
+func (e *DuplicatePostConditionsError) Error() string {
+	return "unexpected second post-conditions"
+}
+
+func (*DuplicatePostConditionsError) SecondaryError() string {
+	return "transaction declarations can only have one 'post' block"
+}
+
+func (*DuplicatePostConditionsError) DocumentationLink() string {
+	return "https://cadence-lang.org/docs/language/transactions"
+}
+
+// ExpectedPrepareOrExecuteError is reported when a 'prepare' or 'execute' block is expected in a transaction.
+type ExpectedPrepareOrExecuteError struct {
+	GotIdentifier string
+	Pos           ast.Position
+}
+
+var _ ParseError = &ExpectedPrepareOrExecuteError{}
+var _ errors.UserError = &ExpectedPrepareOrExecuteError{}
+var _ errors.SecondaryError = &ExpectedPrepareOrExecuteError{}
+var _ errors.HasDocumentationLink = &ExpectedPrepareOrExecuteError{}
+
+func (*ExpectedPrepareOrExecuteError) isParseError() {}
+
+func (*ExpectedPrepareOrExecuteError) IsUserError() {}
+
+func (e *ExpectedPrepareOrExecuteError) StartPosition() ast.Position {
+	return e.Pos
+}
+
+func (e *ExpectedPrepareOrExecuteError) EndPosition(memoryGauge common.MemoryGauge) ast.Position {
+	length := len(e.GotIdentifier)
+	return e.Pos.Shifted(memoryGauge, length-1)
+}
+
+func (e *ExpectedPrepareOrExecuteError) Error() string {
+	return fmt.Sprintf(
+		"unexpected identifier, expected keyword %q or %q, got %q",
+		KeywordPrepare,
+		KeywordExecute,
+		e.GotIdentifier,
+	)
+}
+
+func (*ExpectedPrepareOrExecuteError) SecondaryError() string {
+	return "the first block in a transaction declaration must be a 'prepare' or an 'execute' block"
```

Comment:
Just a nit: the primary error message places `execute` and `prepare` keywords within double-quotes, where as this secondary error message places them within single-quote. Might be good to use the one approach for both.

Also, given there are mixed usages of `%s` and `%q` in `Spintf`, wonder whether using backticks (`%#q`) for all cases might be a better option.